modules/helm/format.go

```diff
@@ -0,0 +1,81 @@
+package helm
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/gruntwork-io/gruntwork-cli/collections"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// FormatSetValuesAsArgs formats the given values as command line args for helm using set (e.g of the format
+// --set key=value)
+func FormatSetValuesAsArgs(setValues map[string]string) []string {
+	args := []string{}
+
+	// To make it easier to test, go through the keys in sorted order
+	keys := collections.Keys(setValues)
+	for _, key := range keys {
+		value := setValues[key]
+		argValue := fmt.Sprintf("%s=%s", key, value)
+		args = append(args, "--set", argValue)
+	}
+
+	return args
+}
+
+// FormatValuesFilesAsArgs formats the given list of values file paths as command line args for helm (e.g of the format
+// -f path). This will fail the test if one of the paths do not exist.
+func FormatValuesFilesAsArgs(t *testing.T, valuesFiles []string) []string {
+	args, err := FormatValuesFilesAsArgsE(t, valuesFiles)
+	require.NoError(t, err)
+	return args
+}
+
+// FormatValuesFilesAsArgsE formats the given list of values file paths as command line args for helm (e.g of the format
+// -f path)
+func FormatValuesFilesAsArgsE(t *testing.T, valuesFiles []string) ([]string, error) {
```

Comment:
What is the difference between this method and `FormatSetFilesAsArgsE`? Is it just the argument (`-f` vs `--set-file`)?

Reply:
There's one more difference, where `--set-file` is key value pairs.

If you want to know what these args are: `-f`, which is short for `--values`, passes in a `values.yaml` file which is a yaml file listing out all the inputs and replaces `--set`. E.g let's say you have

```yaml
# example.yaml
containerImage:
  repo: nginx
  tag: 1.15.4
```

the following are equivalent:

```
helm install -f example.yaml
helm install --set containerImage.repo=nginx --set containerImage.tag=1.15.4
```

`--set-file` is a special version of `--set`, where the value is pulled from a file as opposed to passed in. Let's say that you also had the file `containerrepo.txt` which contains `nginx`. Then the following is also equivalent:

```
helm install --set-file containerImage.repo=./containerrepo.txt --set containerImage.tag=1.15.4
```